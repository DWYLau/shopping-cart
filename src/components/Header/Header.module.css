@import url("https://fonts.googleapis.com/css2?family=Molengo&display=swap");

.header {
  color: white;
  background: #2d2d2d;
  font-family: "Molengo", sans-serif;
  font-weight: 400;
  font-size: 2rem;
  font-style: normal;
  height: 8vh;
  width: 100vw;
  margin-top: 1.5rem;
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 1.5rem;
}

.logo {
  cursor: pointer;
  transition: ease 0.5s;
  padding: 0.5rem;
  border-radius: 0.5rem;
  color: white;
  text-decoration: none;
}

.logo:hover {
  background-color: #c4dfdf;
  box-shadow: 0 0 1rem #c4dfdf;
  color: black;
}

.link-container {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  text-decoration: none;
  text-transform: uppercase;
  font-size: 1.5rem;
  color: white;
}

.link:first-child {
  display: flex;
  align-items: center;
  text-decoration: none;
  text-transform: uppercase;
  font-size: 1.5rem;
  color: white;
  padding: 0 1rem;
  height: 100%;
  border-left: 1px solid white;
  border-right: 1px solid white;
  transition: ease 0.5s;
  letter-spacing: 0.1rem;
}

.link:nth-child(2) {
  display: flex;
  align-items: center;
  text-decoration: none;
  text-transform: uppercase;
  font-size: 1.5rem;
  color: white;
  padding: 0 1rem;
  height: 100%;
  border-right: 1px solid white;
  transition: ease 0.5s;
  letter-spacing: 0.1rem;
}

.link:hover {
  background-color: #c4dfdf;
  box-shadow: 0 0 1rem #c4dfdf;
  color: black;
}

.search-field {
  display: flex;
  justify-content: center;
  align-items: center;
  font-family: "Molengo", sans-serif;
  height: 40%;
  width: 30vw;
  background: white;
  border-radius: 30px;
  transition: ease 0.5s;
  padding: 0.5rem 0;
  padding-left: 0.2rem;
}

.search-field:hover {
  box-shadow: 0 0 1rem white;
}

.input {
  border: none;
  outline: none;
  font-size: 1.3rem;
  width: 100%;
  border: 2px solid white;
  border-radius: 20px;
  transition: ease 0.1s;
  padding: 0.4rem;
}

.input:focus {
  border: 2px solid black;
}

.search-icon {
  cursor: pointer;
  height: 30px;
  width: 30px;
  transition: ease 0.5s;
  border-radius: 50%;
  border: 3px solid white;
  padding: 0.2rem;
}

.search-icon:hover {
  border: 3px solid #c4dfdf;
  border-radius: 50%;
  box-shadow: 0 0 1rem #c4dfdf;
  color: black;
}

.cart-info {
  display: flex;
  align-content: center;
  justify-content: center;
}

.shopping-icon {
  cursor: pointer;
  height: 40px;
  width: 40px;
  margin-right: 1rem;
  transition: ease 0.5s;
  border-radius: 0.5rem;
  padding: 0.2rem;
}

.shopping-icon:hover {
  background-color: #c4dfdf;
  box-shadow: 0 0 1rem #c4dfdf;
  color: black;
}

.cart-number {
  display: flex;
  align-items: center;
  align-content: center;
  justify-content: center;
  background-color: yellow;
  height: 18px;
  width: 18px;
  border-radius: 50%;
  color: black;
  padding: 0.5rem;
  margin-top: 0.5rem;
  font-size: 1.6rem;
  transition: ease 0.5s;
  font-family: system-ui, sans-serif;
  text-decoration: none;
}

.cart-number:hover {
  cursor: pointer;
  box-shadow: 0 0 1rem #c4dfdf;
}

.second-header {
  font-family: "Molengo", sans-serif;
  font-weight: 700;
  font-size: 1.2rem;
  font-style: normal;
  text-transform: uppercase;
  display: flex;
  width: 100%;
}

.banner:first-child {
  background: red;
}

.banner {
  display: flex;
  justify-content: center;
  align-items: center;
  background: #f8f8f8;
  width: 50vw;
  padding: 1rem;
  box-sizing: border-box;
  float: left;
  letter-spacing: 2px;
}

@media (-webkit-device-pixel-ratio: 1.25) {
  .search-field {
    height: 50%;
  }

  .header {
    padding: 0.3rem;
  }
}
