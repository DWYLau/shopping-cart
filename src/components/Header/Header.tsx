import { Link } from "react-router-dom"
import styles from "./Header.module.css"
import Search from "../../assets/icons/search.svg"
import Shopping from "../../assets/icons/shopping-bag.svg"
import { useSearch } from "../context/SearchContext"
import { useCart } from "../context/CartContext"

function Header() {
  const { searchValue, setSearchValue } = useSearch()
  const { cart } = useCart()

  function handleSearch(event: React.ChangeEvent<HTMLInputElement>): void {
    setSearchValue(event.target.value)
  }

  function resetSearch() {
    setSearchValue("")
  }

  return (
    <>
      <header className={styles.header}>
        <Link onClick={resetSearch} className={styles.logo} to={"/"}>
          SHOPPIT
        </Link>
        <div className={styles["link-container"]}>
          <Link onClick={resetSearch} className={styles.link} to="/">
            Home
          </Link>
          <Link onClick={resetSearch} className={styles.link} to="products">
            Products
          </Link>
        </div>
        <div className={styles["search-field"]}>
          <input
            className={styles.input}
            type="search"
            placeholder="Search for items"
            value={searchValue ?? ""}
            onChange={handleSearch}
          />
          <img className={styles["search-icon"]} src={Search} />
        </div>
        <div className={styles["cart-info"]}>
          <Link onClick={resetSearch} to="cart">
            <img
              className={styles["shopping-icon"]}
              src={Shopping}
              alt="Shopping Bag"
            />
          </Link>
          {cart && <div className={styles["cart-number"]}>{cart.length}</div>}
        </div>
      </header>

      <header className={styles["second-header"]}>
        <div className={styles.banner}>50% Global Discount</div>
        <div className={styles.banner}>Free Worldwide Delivery</div>
      </header>
    </>
  )
}

export default Header
