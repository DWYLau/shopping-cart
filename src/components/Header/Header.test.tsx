import { getByPlaceholderText, getByText, render } from "@testing-library/react"
import Header from "./Header"
import { describe, it, expect } from "vitest"
import { BrowserRouter } from "react-router-dom"

describe("Header component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    )
    expect(container).toMatchSnapshot()
  })

  it("renders shoppit", () => {
    const { container } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    )
    getByText(container, "shoppit", { exact: false })
  })

  it("renders an input element", () => {
    const { container } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    )
    getByPlaceholderText(container, "Search for items")
  })
})
