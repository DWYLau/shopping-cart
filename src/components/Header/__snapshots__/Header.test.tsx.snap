// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > renders correctly 1`] = `
<div>
  <header
    class="_header_63c2ea"
  >
    <a
      class="_logo_63c2ea"
      href="/"
    >
      SHOPPIT
    </a>
    <div
      class="_link-container_63c2ea"
    >
      <a
        class="_link_63c2ea"
        href="/"
      >
        Home
      </a>
      <a
        class="_link_63c2ea"
        href="/products"
      >
        Products
      </a>
    </div>
    <div
      class="_search-field_63c2ea"
    >
      <input
        class="_input_63c2ea"
        placeholder="Search for items"
        type="search"
        value=""
      />
      <img
        class="_search-icon_63c2ea"
        src="/src/assets/icons/search.svg"
      />
    </div>
    <div
      class="_cart-info_63c2ea"
    >
      <a
        href="/cart"
      >
        <img
          alt="Shopping Bag"
          class="_shopping-icon_63c2ea"
          src="/src/assets/icons/shopping-bag.svg"
        />
      </a>
      <a
        class="_cart-number_63c2ea"
        href="/cart"
      >
        0
      </a>
    </div>
  </header>
  <header
    class="_second-header_63c2ea"
  >
    <div
      class="_banner_63c2ea"
    >
      50% Global Discount
    </div>
    <div
      class="_banner_63c2ea"
    >
      Free Worldwide Delivery
    </div>
  </header>
</div>
`;
